tests/unit/timer_test.cc

```diff
@@ -92,8 +99,29 @@ struct timer_test {
         return pr2.get_future().then([&t1] { delete &t1; });
     }
 
+    future<> test_cancel_in_periodic() {
+        fmt::print("=== Start test_cancel_in_periodic test\n");
+        t6.set_callback([this](){
+            if (!t6.cancel()) {
+                BUG();
+            }
+            pr3.set_value();
+        });
+        t7.set_callback([this] { OK(); pr4.set_value(); });
+        t6.arm_periodic(10ms);
+        (void)pr3.get_future().then([this]{
+            t7.arm(100ms);
+        });
+
+        //some long 3rd party unstoppable calculation
+        std::this_thread::sleep_for(100ms);
```

Comment:
I don't understand this comment. Why do you need the sleep_for()? the waiting for p4's future will wait for more than 100ms anyway.
Or maybe what you're saying is that somehow the cancel() broke this sleep_for() and before this patch, the sleep_for would never finish? If this is the case, please a comment saying this, and not some statement about "long 3rd party unstoppable calculation", whatever that means.

Reply:
Added explicit comment that this line is for reactor delay